zero_bin/common/src/provider.rs

```diff
@@ -8,7 +8,7 @@ use anyhow::Context;
 use tokio::sync::{Mutex, Semaphore, SemaphorePermit};
 
 const CACHE_SIZE: usize = 1024;
-const MAX_NUMBER_OF_PARALLEL_REQUESTS: usize = 64;
+const MAX_NUMBER_OF_PARALLEL_REQUESTS: usize = 128;
```

Comment:
technically not just a trivial move to `common`, but I trust that you properly measured this to be satisfactory :)